docs/proposals/registry/registry-structure.md

```diff
@@ -0,0 +1,101 @@
+
+## Problem
+Our approach of pushing only the devfile.yaml works fine currently because the stacks we’re pushing only have a devfile.yaml in them. However, devfile stacks may contain more than _just_ the devfile, and may also contains resources that are re-used across multiple stacks (such as certain vsx plugins).
+
+We need a defined way of knowing what files in a stack to push up in a, what layers each file in the stack should belong to, and where the stack's files should located. 
+
+## Proposal
+
+To solve the issue listed above, we should:
+1) Formally the expected structure of the Devfile Regisry Repository.
+2) Define the layers that compose a devfile stack on an OCI registry, and what each layer contains.
+
+### Repository Structure
+The structure of the “Devfile Registry Repository” should impose the following requirements:
+1) A top-level folder called `stacks`, which contains folders for each devfile stack.
+2) Each devfile stack folder must contain a `devfile.yaml`. Other files such as vsx plugins, stack logos, etc. can be included as needed.
+3) A Dockerfile to package the stacks and index.json (see https://github.com/devfile/api/blob/master/docs/proposals/registry/devfile-packaging.md)
+4) Build scripts or tools for the registry (see https://github.com/devfile/api/blob/master/docs/proposals/registry/devfile-packaging.md)
+
+### Layer Media types
+Currently, when we push devfile stacks to an OCI registry, it's pushed as a single layer, using the `application/vnd.devfileio.devfile.layer.v2+yaml` media type. We should instead be pushing the stack as a multi-layer artifact, adding the additional layers:
+
+**VSX Plugins**
+
+`*.vsx` -> `application/vnd.devfileio.vsx.layer.v1.tar`
+
+**Stack logos**
+
+`logo.svg` -> `image/svg+xml` or
+
+`logo.png` -> `image/png`
+
+**Everything else**
+
+`archive.tar` - `application/x-tar`
+
+As part of the registry build process that packages the stacks into a container image, any files not belonging to the devfile, vsx or logo media types will be lumped together in a tar archive (using the media type `application/x-tar`).
+
+When the registry bootstrap process pushes the stack up to the OCI registry, each file belonging to one of the above media types (`devfile.yaml`, `*.vsx`, `logo.svg`/`logo.png`, `archive.tar`) will be treated as separate layers in the artifact.
+
+
+### Index.json Modification
+The index.json already includes a link to a stack’s devfile, which the registry bootstrap process parses when pushing devfiles up to the registry. 
+
+To make it easier to bootstrap the registry, and to avoid having to programmatically find which files to push individually; as part of the registry build process, include a new `layers` array in the index.json that tells the registry bootstrap process which layers the stack is composed of:
```

Comment:
Should we call this `resources` or `artifacts` instead of `layers`?

Reply:
I went with `layers` since that's what OCI uses, and since we'll probably need to use the files specified in `layers` when we implement caching on a per-layer basis.

But it is just a name, and I'm flexible either way with `resources` or `layers`.

Reply:
Updated to change to `resources`.